docs/posts/ibis-duckdb-geospatial/index.qmd

```diff
@@ -0,0 +1,182 @@
+
+::: {.callout-note}
+You can check Dr. Qiusheng Wu's full Spatial Data Management course material in its  [website](https://geog-414.gishub.org/index.html), and the classes are also on [youtube](https://www.youtube.com/watch?v=A4TOAdsXsEs&list=PLAxJ4-o7ZoPe9SkgnophygyLjTDBzIEbi),
+:::
+
+## Setup and Data
+
+At the moment, the geospatial features for the DuckDB backend are available on the main branch of Ibis, but they'll be included in the next release.
+
+
+```{python}
+from pathlib import Path
+from zipfile import ZipFile
+from urllib.request import urlretrieve
+
+# Download and unzip
+url = "https://open.gishub.org/data/duckdb/nyc_data.db.zip"
+zip_path = Path("nyc_data.db.zip")
+db_path = Path("nyc_data.db")
+
+if not zip_path.exists():
+    urlretrieve(url, zip_path)
+
+if not db_path.exists():
+    with ZipFile(zip_path) as zip_file:
+        zip_file.extract('nyc_data.db')
+```
+
+## Let's get started
+
+The beauty about spatial databases is that they not only allow us to store geometries in them, but also compare them.
+
+```{python}
+import ibis
+from ibis import _
+
+ibis.options.interactive = True
+
+con = ibis.duckdb.connect("nyc_data.db")
+con.list_tables()
+```
+
+In this database we have multiple tables with information about New York City. Following, Dr. Qiusheng Wu's class, we'll take a look at some spatial relations.
+
+We can start by taking a peak at the `nyc_subway_stations` table.
+
+```{python}
+subway_stations = con.table('nyc_subway_stations')
+subway_stations
+```
+
+Notice that the last column has a `geometry` type, and in this case it contains a points that represent the location of each subway station. Let's take the entry for `'Broad St'`.
+
+```{python}
+broad_station =  subway_stations.filter(subway_stations.NAME == 'Broad St')
+broad_station
+```
+
+### `geo_equals` (`ST_Equals`)
+
+In DuckDB `ST_Equals` returns `True` if two geometries are topologically equal, meaning that they have the same dimension and they have identical coordinate values, although the order of the vertices may be different.
+
+The following  is a bit redundant but we can check if our broad station point
+matches only one point on our data using `geo_equals`
+
+```{python}
+subway_stations.filter(subway_stations.geom.geo_equals(broad_station.geom))
+```
+
+We can also write this query without using `broad_station` as a variable, and with the help of the deferred expressions API, also known as Underscore (`_`) API (Learn more about it in the [chaining expressions docs](https://ibis-project.org/how-to/analytics/chain_expressions.html))
```

Comment:
```suggestion
We can also write this query without using `broad_station` as a variable, and with the help of the deferred expressions API, also known as [the underscore API](https://ibis-project.org/how-to/analytics/chain_expressions.html).
```

No need to mention it three times (deferred, underscore and then link to the docs). We can link directly from the mention.